app/services/immunization_service/follow_up.rb

```diff
@@ -86,23 +86,27 @@ def process_vaccine_schedules(vaccine_schedules, client_missed_visits, under_fiv
       vaccine_schedules.each do |vaccine_schedule|
         vaccine_schedule[1].each do |visit|
 
-          missed_antigens = visit[:antigens].select { |antigen| antigen[:can_administer] && antigen[:status] == "pending" && visit[:milestone_status] == "current" }
+          missed_antigens = visit[:antigens].select { |antigen| antigen[:can_administer] && antigen[:status] == "pending" && visit[:milestone_status] == "passed" }
 
           unless missed_antigens.empty?
             client_missed_visits << { visit: visit[:visit], milestone_status: visit[:milestone_status], age: visit[:age], antigens: missed_antigens }
             update_missed_doses(missed_antigens, immunization_client.birthdate, under_five_missed_doses, over_five_missed_doses)
 
             # Calculate if the vaccine can be admnistered and its pending 
+          end
+
+          due_antigens = visit[:antigens].select { |antigen| antigen[:can_administer] && antigen[:status] == "pending" && visit[:milestone_status] == "current"}
+
+          unless due_antigens.empty?
             due_date = calculate_due_date(immunization_client.birthdate, visit[:age])
```

Comment:
I am thinking if instead of shifting in calculate_due_date if we would add and seems you might have also missed the months in month calculation